internal/graphapi/orgextended.resolvers.go

```diff
@@ -56,6 +59,124 @@ func (r *mutationResolver) CreateOrganizationWithMembers(ctx context.Context, or
 	}, nil
 }
 
+// TransferOrganizationOwnership is the resolver for the transferOrganizationOwnership field.
+func (r *mutationResolver) TransferOrganizationOwnership(ctx context.Context, organizationID string, newOwnerID string) (*model.OrganizationTransferOwnershipPayload, error) {
+	// Step 1: Get current user from context
+	currentUserID, err := auth.GetSubjectIDFromContext(ctx)
+	if err != nil {
+		log.Error().Err(err).Msg("unable to get current user from context")
+		return nil, ErrInternalServerError
+	}
+
+	c := withTransactionalMutation(ctx)
+
+	// Step 2: Verify organization exists
+	org, err := c.Organization.Get(ctx, organizationID)
+	if err != nil {
+		return nil, parseRequestError(err, action{action: ActionGet, object: "organization"})
+	}
+
+	// Step 3: Verify current user is the owner
+	currentUserMembership, err := c.OrgMembership.Query().
+		Where(
+			orgmembership.OrganizationID(organizationID),
+			orgmembership.UserID(currentUserID),
+		).
+		Only(ctx)
+	if err != nil {
+		log.Error().Err(err).Msg("unable to find current user's membership")
+		return nil, newNotFoundError("org_membership")
+	}
+
+	if currentUserMembership.Role != enums.RoleOwner {
+		log.Info().
+			Str("user_id", currentUserID).
+			Str("organization_id", organizationID).
+			Str("role", currentUserMembership.Role.String()).
+			Msg("user attempted to transfer ownership without being owner")
+		return nil, newPermissionDeniedError()
+	}
+
+	// Step 4: Verify new owner user exists
+	allowCtx := privacy.DecisionContext(ctx, privacy.Allow)
+	newOwnerUser, err := c.User.Get(allowCtx, newOwnerID)
+	if err != nil {
+		log.Error().Err(err).Str("new_owner_id", newOwnerID).Msg("new owner user not found")
+		return nil, newNotFoundError("user")
+	}
+
+	// Step 5: Check if new owner is already a member
+	newOwnerMembership, err := c.OrgMembership.Query().
+		Where(
+			orgmembership.OrganizationID(organizationID),
+			orgmembership.UserID(newOwnerID),
+		).
+		First(ctx)
```

Comment:
I think `Only` is safer, there shouldn't ever be duplicates if DeletedAtIsNil is added to the `Where`
```suggestion
		Only(ctx)
```